scripts/cloud_masking.py

```
@@ -0,0 +1,151 @@
+        '''
+
+        (status, output) = commands.getstatusoutput(cmd_string)
+
+        if status < 0:
+            message = ('Application terminated by signal [{0}]'
+                       .format(cmd_string))
+            if len(output) > 0:
+                message = ' Stdout/Stderr is: '.join([message, output])
+            raise ExecuteError(message)
+
+        if status != 0:
+            message = 'Application failed to execute [{0}]'.format(cmd_string)
+            if len(output) > 0:
+                message = ' Stdout/Stderr is: '.join([message, output])
+            raise ExecuteError(message)
+
+        if os.WEXITSTATUS(status) != 0:
+            message = ('Application [{0}] returned error code [{1}]'
+                       .format(cmd_string, os.WEXITSTATUS(status)))
+            if len(output) > 0:
+                message = ' Stdout/Stderr is: '.join([message, output])
+            raise ExecuteError(message)
+
+        return output
+
+
+def parse_cmd_line():
+        '''Will only parse --xml XML_FILENAME from cmdline.
+
+        Precondition:
+            '--xml FILENAME' exists in command line arguments
+        Postcondition:
+            returns xml_filename
+
+        Note: Help is not included because the program will return
+              the help from the underlying program.
+        '''
+
+        # Try to parse out the XML so the application can be determined
+        parse_xml = argparse.ArgumentParser(add_help=False)
+        parse_xml.add_argument('--xml', action='store',
+                               dest='xml_filename', required=True,
+                               help='Input XML metadata file',
+                               metavar='FILE')
+        (temp, extra_args) = parse_xml.parse_known_args()
+
+        return temp.xml_filename
+
+
+def get_science_application_name(satellite_sensor_code):
+    '''Returns name of executable that needs to be called'''
+
+    l8_prefixes = ['LC8', 'LO8']
```

We are actually just letting the OLI only fall through to the C application and fail there for consistent error messaging.
The idea was to have very little smarts, only enough to call the correct C application when they are different for different sensors.

I'll add the LT8 and let it fail like OLI only in the C application.